Release notes v1.61.0 (#658)
replicatedhq/kots.io: content/release-notes/1.61.0.md

---
date: 2022-01-28T10:21:28-07:00
linktitle: "1.61.0"
title: "1.61.0"

> kubernetes: "1.20, 1.21, and 1.22"

Reviewer: We're supporting four versions right now since all four of these are in kURL and are [officially in support](https://kubernetes.io/releases/).
```suggestion
kubernetes: "1.20, 1.21, 1.22, and 1.23"
```

---

{{<features>}}
* Adds a CLI command to collect all versions for an application. See [the docs](https://kots.io/kots-cli/get/versions/) for more details.

> * Adds the ability to block installing a certain application if the current KOTS version is incompatible with the KOTS version required by the app.
> * Adds the ability to block upgrading the application if the current KOTS version is incompatible with the KOTS version required by the release.

Reviewer: I agree with a comment @JustinN-Replicated made in Slack that these could be combined into one line. Maybe like this.
```suggestion
* Adds the ability to block installing or upgrading an application if the current KOTS version is incompatible with the KOTS version required by that version of the application.
```

Reviewer 2: lgtm

---
date: 2022-01-28T10:21:28-07:00
linktitle: "1.61.0"
title: "1.61.0"
kubernetes: "1.20, 1.21, and 1.22"
---

{{<features>}}

> * Adds a CLI command to collect all versions for an application. See [the docs](https://kots.io/kots-cli/get/versions/) for more details.

Reviewer 2: Do we usually abbreviate as 'the docs'?
Suggest the following:
Adds a CLI command to collect all available versions for an application from app manager. See [the documentation](https://kots.io/kots-cli/get/versions/) for more details.

Reviewer: Even shorter, we could hyperlink the CLI command and get rid of the second sentence altogether. (I went with KOTS since we haven't really rebranded the docs to use app manager yet.) And "get" sounds more natural than "collect", especially since the command name looks like `get` (inferred from the docs link).
```suggestion
* Adds a CLI command to [get all available versions](https://kots.io/kots-cli/get/versions/) for an application from KOTS.
```

Reviewer 2: Definitely good recommendations; lgtm

* Adds the ability to block installing a certain application if the current KOTS version is incompatible with the KOTS version required by the app.
* Adds the ability to block upgrading the application if the current KOTS version is incompatible with the KOTS version required by the release.

> * Adds functionality to display a modal with commands for upgrading KOTS if there the version attempting to be deployed has a different KOTS version specified than the one the user has installed.

Reviewer 2: I think this is supposed to be the following:
Adds functionality to display a modal with commands for upgrading KOTS if the version attempting to be deployed has a different KOTS version specified than the one the user has installed.

Reviewer 3: this release note is no longer accurate. let's remove it

{{</features>}}

{{<changes>}}

{{</changes>}}

{{<fixes>}}
* Fixes a bug that caused images to be pushed to private registry multiple times during airgap install.

> * Fixes a bug that displayed a message to "edit the current config when performing a new install

Reviewer 2: I don't think the quotes are neceesary here (they also do not close or have proper punctuation to end the sentence)I think this should be something more like the following:
* Fixes a bug that erroneously displays a message to edit the current config when performing a new install.

Alternatively:
* Fixes a bug that erroneously displays a message to edit the current config when performing a new install where a current config does not exist.

> * Fixes an issue that caused [image garbage collection](/kotsadm/registries/kurl-registry/#image-garbage-collection) to not delete images with tags other than "latest."

Reviewer 2: This could likely be shortened: 
* Fixes an issue that caused [image garbage collection](/kotsadm/registries/kurl-registry/#image-garbage-collection) to only remove images with the "latest" tag.
